### PR TITLE
Improve 'solana stakes' cli

### PR DESCRIPTION
**Pull request by Hassall**
#### Problem
- No error message returned
- Does not accept identity

#### Summary of Changes
- ~~Remove filter that would only show stakes for  `vote_pubkey`~~
- Show stakes for identity/vote pubkey

Looking for feedback @mvines 

#### ~~Observed Issue~~
- ~~Although the CLI accepts multiple values for `solana stakes`, the array, `vote_account_pubkeys`,  will always consist of just the first value.~~
~~_I think it's for the best that only one key is accepted at a time._~~

Fixes 
- https://github.com/solana-labs/solana/issues/8441

### CODE REVIEW

File: cli/src/cluster_query.rs [hunk 1]
```diff
@@ -1693,39 +1692,51 @@ pub fn process_show_stakes(
...
+            pubkeys.remove(value);
+        }
+        return Err(format!(
+            "Failed to retrieve some or all stake accounts: {:?}.",
+            pubkeys
+        )
+        .into());
+    }
+
+    let all_stake_accounts = rpc_client.get_program_accounts(&stake::program::id())?;
```
Comment (mvines) [inline cli/src/cluster_query.rs:1739]: Dropping the server-side filter when only one stake account is requested is not great for user and RPC server performance.  Any way to bring that back here?

Comment (Hassall) [reply cli/src/cluster_query.rs:1739]: sure, i can take a look and see if i can modify the filter to allow for `node_pubkey`/`identity` and not just `vote_pubkey`

File: cli/src/cluster_query.rs [hunk 2]
```diff
@@ -1701,28 +1723,24 @@ pub fn process_show_stakes(
...
+                ),
+                encoding: Some(rpc_filter::MemcmpEncoding::Binary),
+            }),
+            // Filter by `Delegation::voter_pubkey`, which begins at byte offset 124
+            rpc_filter::RpcFilterType::Memcmp(rpc_filter::Memcmp {
+                offset: 124,
+                bytes: rpc_filter::MemcmpEncodedBytes::Base58(vote_pubkey.clone()),
+                encoding: Some(rpc_filter::MemcmpEncoding::Binary),
+            }),
+        ]);
```
Comment (diman-io) [inline cli/src/cluster_query.rs:1743]: If you use the shadowing you'll not need to change this block.

File: cli/src/cluster_query.rs [hunk 3]
```diff
@@ -1757,9 +1775,7 @@ pub fn process_show_stakes(
                 }
                 StakeState::Stake(_, stake) => {
                     if vote_account_pubkeys.is_none()
-                        || vote_account_pubkeys
-                            .unwrap()
-                            .contains(&stake.delegation.voter_pubkey)
+                        || vote_pubkey.eq(&stake.delegation.voter_pubkey.to_string())
```
Comment (diman-io) [inline cli/src/cluster_query.rs:1778]: ... and this too.

**Comment by diman-io** [comment]
Hey @Hassall 
By "shadowing" I meant something like this
https://gist.github.com/diman-io/988778f83fb14b5dee56f31d4524d476

**Pull request closed without merging**